hbase-rest/src/main/java/org/apache/hadoop/hbase/rest/client/Client.java

```diff
@@ -970,4 +986,18 @@ public ClientTrustStoreInitializationException(String message, Throwable cause)
       super(message, cause);
     }
   }
+
+  public void close() {
+    try {
+      // We may not have set an explicit connection manager, so we need to use this.
+      httpClient.getConnectionManager().shutdown();
+    } catch (Exception e) {
+      LOG.info("Exception while shutting down connection manager", e);
+    }
+  }
+
+  @Override
+  public void finalize() {
```

Comment:
Seems it is not suggested to use finalize in modern java application. Is this necessary?

Reply:
Realistically, when this object gets GCd, the underlying apache httpClient / pool object will also get GCd, and those should take care of closing themselves properly in its their finalizer, so it is not strictly necessary.

However, I think that closing resources like tcp connections is a valid use case for finalizers, and it's better to clean up explicitly than to rely on the lower layers' finalizers.

Reply:
I have done some googling, almost all the related articles highly suggest to not use this feature any more, and the finalize method has been deprecated in java 9.

For me, if there are no strong reasons, we'd better not use this method any more.

Reply:
OK, Done.